azext_iot/operations/_mqtt.py

```diff
@@ -5,95 +5,76 @@
-        six.print_(
-            "Connected to target IoT Hub MQTT broker with result: {}".format(
-                connection_result[rc]
-            )
-        )
-        self.client.subscribe(self.topic_receive)
-        six.print_("Subscribed to device bound message queue")
-        self.connected = True
-
-    def on_message(self, client, userdata, msg):
+        self.registry_manager = IoTHubRegistryManager(target["cs"]) 
+        self.device_client = mqtt_device_client.create_from_connection_string(device_connection_string)
+        self.digital_twin_client =  DigitalTwinClient(target["cs"])
+        self.device_client.connect()
+        self.device_client.on_message_received = self.message_handler
+        self.device_client.on_method_request_received = self.method_request_handler
+
+    def send_c2d_message(self, message_text="", properties={}):
+        result = {
+            "status": None,
+            "error": None
+        }
+        try:
+            self.registry_manager.send_c2d_message(self.device_id, message_text, properties)
+            result.status = "C2D Message sent succesfully"
+        except Exception as ex:
+            result.status = "Failed to send C2D Message"
+            result.error = "Unexpected error {0}".format(ex)
+        return result
+    
+    def send_d2c_message(self, message_text="", properties={}):
+        message = Message(message_text)
+        message.custom_properties = properties
+        self.device_client.send_message(message)
+    
+    def invoke_direct_method(self, command_name, payload="", connect_timeout_in_seconds=100, response_timeout_in_seconds=100):
+        self.digital_twin_client.invoke_command(self.device_id, command_name, payload, connect_timeout_in_seconds, response_timeout_in_seconds)
+
+    def message_handler(self, message):
         six.print_()
-        six.print_("_Received C2D message with topic_: {}".format(msg.topic))
-        six.print_("_Payload_: {}".format(msg.payload))
-
-    def on_publish(self, client, userdata, mid):
-        six.print_(".", end="", flush=True)
-
-    def is_connected(self):
-        return self.connected
-
-    def execute(self, data, publish_delay=2, msg_count=100):
+        six.print_("_Received C2D message with topic_: /devices/{}/messages/devicebound".format(self.device_id))
+        six.print_("_Payload_: {}".format(message.data))
+        if message.custom_properties:
+            six.print_("_Custom Properties_: {}".format(message.custom_properties))
+    
+    def method_request_handler(self, method_request):
+        six.print_()
+        six.print_("Received method request with id: '{}' and method name: '{}' for device with id: '{}'".format(method_request.request_id, method_request.name, self.device_id))
+        six.print_("_Payload_: {}".format(method_request.payload))
+        payload = {"result": True, "data": "Method succeeded"}  # set response payload
+        status = 200  # set return status code
```

Comment:
we don't want to always hardcode 200 for the status....both the response code and response payload should have defaults, but can be customized/overridden by user.